spec/README.md

```diff
@@ -0,0 +1,172 @@
+
+The pool has two main components:
+* **oracle**: Calculates off-chain the rewards that each participant of the smoothing pool can claim, using on-chain data from the consensus and execution layer. It also summarizes all balances and addresses with the right to claim in a merkle root, that is periodically stored on-chain, making it non-revertable. On the other hand it serves as a data availability layer, that can be used to get the proofs one needs to use on-chain to claim their share of the rewards.
+* **contract**: Gets all the rewards produced by the members of the pool and provides a set of functions to subscribe, unsubscribe and claim rewards by providing the appropriate proofs of belonging to the merkle tree.
+
+## Configuration parameters
+
+The oracle shall contain the following configuration parameters, that will determine how rewards are calculated:
+* `COLLATERAL_GWEI`: Amount of collateral in gwei that a validator need to join the pool.
+* `DEPLOYED_SLOT`: Slot when the smoothing pool contract was deployed.
+* `CHECKPOINT_SIZE_SLOTS`: How often the smart contract root is updated with new rewards, in slots.
+* `OWNER_ADDRESS`: Address with `0x` prefix of the account allowed to update the smoothing pool contract.
+* `POOL_CONTRACT_ADDRESS`: Address with `0x` prefix of the smoothing pool contract.
+* `NETWORK`: Network where rewards are being calculated: `mainnet` or `goerli`.
+* `POOL_FEES_ADDRESS`: Address with `0x` prefix of the account that can claim the smoothing pool fees.
+* `POOL_FEES_PERCENT`: Amount in % that `POOL_FEES_ADDRESS` gets for every reward sent to the smoothing pool. Note that it also gets rounding remainders on top, but this is almost neglectable.
+
+## Source of rewards
+
+A `Reward` is considered to be any balance denominated in Eth that is sent to the `POOL_CONTRACT_ADDRESS`. These are detected by the oracle and shared fairly among all the participants in the pool at a given time. The oracle shall detect all these types of rewards and distribute them fairly (see rewards calculation section). All of these rewards are denominated in `ETH` and other types of tokens such as ERC20 are not considered by the oracle:
+* `MevBlock`: Comes from a block proposal where the reward was obtained via an off-chain agreement using tools such as mev-boost, usually coming as the last transaction in the block.
+* `VanilaBlock`: Comes from a block proposal where the reward was sent via the protocol `fee_recipient`.
+* `PayableDonation`: A donation was made to the smoothing pool contract, sent as a normal transaction detected in the payable.
+* `NonPayableDonation`: A donation was made to the smoothing pool contract, sent via a smart contract, which is not seen in the payable.
+
+## Subscribe/unsubscribe/ban
```

Comment:
self note.

subscribe is lacking few things.

eg: the validator must also not be slashed, exited. must be active!